src/AutoRest.CSharp/Common/Output/Models/PropertyDeclaration.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using AutoRest.CSharp.Common.Output.Expressions.ValueExpressions;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Generation.Writers;
+using AutoRest.CSharp.Output.Models;
+using Azure.Core;
+
+namespace AutoRest.CSharp.Common.Output.Models
+{
+    [DebuggerDisplay("{GetDebuggerDisplay(),nq}")]
+    internal record PropertyDeclaration(FormattableString? Description, MethodSignatureModifiers Modifiers, CSharpType PropertyType, CodeWriterDeclaration Declaration, PropertyAccessorMethod Get, PropertyAccessorMethod? Set = null, ValueExpression? InitializationValue = null)
```

Comment:
One design option I would consider here is to add something like `PropertyBody`, which can be `GetOnlyPropertyBody`, `GetSetPropertyBody`, `ConstantBody`, etc.

Reply:
We still will keep the `PropertyDeclaration` here right? This comment is for the `PropertyAccessorMethod Get, PropertyAccessorMethod? Set = null` part?

I choose this because I am trying to mimic the same way in C# reflection and syntax tree.
And the get and set might have different modifiers, it might be cumbersome to have property body which contains two things.

Reply:
> We still will keep the PropertyDeclaration here right?

Yes

> I choose this because I am trying to mimic the same way in C# reflection and syntax tree.

Reflection represents metadata rather than the way we want to generate the code. Roslyn semantic representation of properties has been created at the times of C# 4.0, and properties have seriously evolved since then, and they have lots of mutually-exclusive conditions that I would prefer to enforce on the level of the type (e.g. initialization value can't co-exist with getter body). What I meant is something like this:

```cs
internal record PropertyDeclaration(FormattableString? Description, MethodSignatureModifiers Modifiers, CSharpType PropertyType, CodeWriterDeclaration Declaration, PropertyBody Body);

internal record PropertyBody();

// Writes `MyProperty { get; set; }` or `MyProperty { get; set; } = InitializationValue;`
internal record TrivialPropertyBody(bool HasSetter, MethodSignatureModifiers SetterModifiers, ConstantExpression? InitializationValue) : PropertyBody;

// Writes `MyProperty { get; }` when GetterBody is null and `MyProperty { get {...} }` when it is present
internal record MethodPropertyBody(MethodBody GetterBody, MethodBody? SetterBody, MethodSignatureModifiers SetterModifiers) : PropertyBody;

// Writes `MyProperty => [[ValueExpression]]
internal record ExpressionPropertyBody(ValueExpression GetterBodyExpression) : PropertyBody;
```

Reply:
I will update accordingly.